packages/components/src/horizontal-bar-list/horizontal-bar-grid-item.tsx

```diff
@@ -18,50 +20,86 @@ const HorizontalBarListItem = ( {
 	useShortLabel,
 	isStatic,
 }: HorizontalBarListItemProps ) => {
-	const { label, value, shortLabel } = data;
+	const { label, value, shortLabel, children: itemChildren } = data;
 	const fillPercentage = maxValue > 0 ? ( value / maxValue ) * 100 : 0;
-	const isLink = url || onClick;
+	const isLink = url || ( onClick && ! itemChildren );
+	const hasChildren = !! itemChildren;
+	const [ open, setOpen ] = useState( false );
+
+	const toggleOpen = ( e: React.MouseEvent | React.KeyboardEvent ) => {
+		e?.preventDefault();
+		setOpen( ! open );
+	};
 
 	const onClickHandler = ( e: React.MouseEvent ) => {
 		e?.preventDefault();
-		onClick?.( e );
+		onClick?.( e, data );
 	};
 
 	const onKeyDownHandler = ( e: React.KeyboardEvent ) => {
 		if ( e?.key !== 'Escape' && e?.key !== 'Tab' && e?.key !== 'Shift' ) {
 			e?.preventDefault();
-			onClick?.( e );
+			onClick?.( e, data );
 		}
 	};
 
 	const TagName = isLink ? 'a' : 'div'; // group parents and countries don't use anchors.
 	const labelText = decodeEntities( useShortLabel ? shortLabel || '' : label ); // shortLabel as an empty string to make TS happy
 
 	return (
-		<li
-			className={ classnames(
-				`${ BASE_CLASS_NAME }-item`,
-				isLink && `${ BASE_CLASS_NAME }-item--link`,
-				hasIndicator && `${ BASE_CLASS_NAME }-item--indicated`,
-				isStatic && `${ BASE_CLASS_NAME }-item--static`
-			) }
-			style={ {
-				[ `--${ BASE_CLASS_NAME }-fill` ]: `${ fillPercentage }%`,
-			} }
-			onClick={ ! url ? onClickHandler : undefined } // only execute onClick if url is not defined, otherwise anchor will be ignored
-			onKeyDown={ ! url ? onKeyDownHandler : undefined }
-		>
-			<div className={ `${ BASE_CLASS_NAME }-item-bar` }>
-				{ leftSideItem && <span>{ leftSideItem }</span> }
-				<TagName className="label" href={ url } tabIndex={ 0 }>
-					{ labelText }
-				</TagName>
-				{ rightSideItem && (
-					<span className={ `${ BASE_CLASS_NAME }--hover-action` }>{ rightSideItem }</span>
+		<>
+			<li
+				className={ classnames(
+					`${ BASE_CLASS_NAME }-item`,
+					isLink && `${ BASE_CLASS_NAME }-item--link`,
+					hasIndicator && `${ BASE_CLASS_NAME }-item--indicated`,
+					isStatic && `${ BASE_CLASS_NAME }-item--static`
 				) }
-			</div>
-			<div className="value">{ value }</div> { /* Add number format */ }
-		</li>
+				style={ {
+					[ `--${ BASE_CLASS_NAME }-fill` ]: `${ fillPercentage }%`,
+				} }
+				onClick={ ! url ? onClickHandler : undefined } // only execute onClick if url is not defined, otherwise anchor will be ignored
+				onKeyDown={ ! url ? onKeyDownHandler : undefined }
+			>
+				<div className={ `${ BASE_CLASS_NAME }-item-bar` }>
+					{ leftSideItem && <span>{ leftSideItem }</span> }
+					<TagName className="label" href={ url } tabIndex={ 0 }>
+						<span>{ labelText }</span>
+						{ hasChildren && (
+							<span role="button" onClick={ toggleOpen } onKeyDown={ toggleOpen } tabIndex={ 0 }>
```

Comment:
Binding toggleOpen to keydown events make it possible to expand/close the sublist with any key presses, including `shift` and `3`, to name a few. We may want to restrict this to `enter` and `space`.

Reply:
That's a great point. It should be addressed now. I used the same mechanism used for tabbing through other interactive items and instead of filtering out some keys I revered the logic only to accept enter and space keys.